### PR TITLE
Northwest Territories Credit

### LINKED ISSUE #229: Northwest Territories Credit
Reference: [https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/northwest-territories/5012-pc/information-residents-northwest-territories.html#P153_16428](https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/northwest-territories/5012-pc/information-residents-northwest-territories.html#P153_16428)

### PR DESCRIPTION
**Pull request by Kenyaokun**
Fixes #229

### CODE REVIEW

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/threshold/high/rate.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: Northwest Territories living tax credit high income rate
+values:
+  2022-12-31: 0.01
+metadata:
+  unit: year
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/threshold/high/rate.yaml:5]: unit: /1

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Northwest Territories Credit cost of living tax credit maximum aount
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:1]: description: Northwest Territories Credit cost of living tax credit maximum amount.

- also add periods in the end of parameter descriptions

File: policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
+from policyengine_canada.model_api import *
+
+
+class nt_living_tax_credit(Variable):
+    value_type = float
+    entity = Person
+    label = "Max for living tax credit"
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py:7]: add province name and full program name

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/threshold/middle/base.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+description: Northwest Territories cost of living tax credit threshold for middle income person in 2022
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/threshold/middle/base.yaml:1]: make the whole table from the [pdf](https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5012-tc/5012-tc-22e.pdf) a marginal rate parameter

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/income_threshold.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+description: Northwest Territories cost of living tax credit income threshold in 2022
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/income_threshold.yaml:1]: make this a single amount parameter with 942 over 66,000 and 0 if below

File: policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
+from policyengine_canada.model_api import *
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py:1]: we can switch to just one file with a marginal rate parameter and single amount for max amount

**Comment by Kenyaokun** [comment]
Request for review

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml [hunk 7]
```diff
@@ -0,0 +1,9 @@
+description: Northwest Territories Credit cost of living tax credit maximum amount in 2022
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:1]:
```suggestion
description: Northwest Territories allows for this maximum amount under the cost of living tax credit.
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+values:
+  2022-12-31: 942 
+metadata:
+  unit: currency-CAD
+  label: Northwest Territories Credit cost of living tax credit maximum amount in 2022
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:6]:
```suggestion
  label: Northwest Territories cost of living tax credit max amount
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+      2022-12-31: 0.0125
+  - threshold:
+      2022-12-31: 48_000
+    rate:
+      2022-12-31: 0.01  
+metadata:
+  type: marginal_rate
+  rate_unit: /1
+  threshold_unit: currency-CAD
+  label: Northwest Territories Cost of Living Credit different phase rate
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml:19]:
```suggestion
  label: Northwest Territories cost of living phase out rate
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+  reference:
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:7]: need to add the northwest territories income tax act as a reference

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+  reference:
+    - title: Government of Canada - Northwest Territories Cost of Living Credit
+      href: https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/northwest-territories/5012-tc.html
+      link: https://www.canada.ca/content/dam/cra-arc/formspubs/pbg/5012-tc/5012-tc-22e.pdf
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml:23]: change format to title and href

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml [hunk 12]
```diff
@@ -0,0 +1,23 @@
+description: Northwest Territories Cost of Living Credit different phase rate and threshold
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml:1]:
```suggestion
description: Northwest Territories extends the following cost of living credit as a fraction of the adjusted net income.
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/reduction.yaml:1]: rename to rate.yaml

File: policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py [hunk 13]
```diff
@@ -0,0 +1,20 @@
...
+    defined_for = ProvinceCode.NT
+
+    def formula(person, period, parameters):
+        p = parameters(
+            period
+        ).gov.provinces.nt.tax.income.credits.living_tax_credit
+
+        net_income = person("individual_net_income", period)
+
+        living_tax_credit = p.reduction.calc(net_income)
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py:18]:
```suggestion
        uncapped_credit = p.reduction.calc(net_income)
```

File: policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py [hunk 14]
```diff
@@ -0,0 +1,20 @@
+from policyengine_canada.model_api import *
+
+
+class nt_living_tax_credit(Variable):
```
Comment (PavelMakarchuk) [inline policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py:4]:
```suggestion
class nt_cost_of_living_tax_credit(Variable):
```

Comment (PavelMakarchuk) [reply policyengine_canada/variables/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_living_tax_credit.py:4]: rename .py and .yaml files

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml [hunk 15]
```diff
@@ -0,0 +1,11 @@
+description: Northwest Territories allows for this maximum amount under the cost of living tax credit.
+values:
+  2022-12-31: 942 
+metadata:
+  unit: currency-CAD
+  label: Northwest Territories cost of living tax credit max amount
+  reference:
+  - title: Government of Canada - Credit
+    href: https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/northwest-territories/5012-tc.html 
+          https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/northwest-territories/5012-pc/information-residents-northwest-territories.html
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:10]: need to change reference to the same format as the first one:
- title:
   href:

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:10]: both yaml files

File: policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml [hunk 16]
```diff
@@ -0,0 +1,63 @@
+- name: Income below 12_000
+  period: 2023
+  input:
+    province_code: NT
+    individual_net_income: 12_000
+  output:
+    nt_cost_of_living_tax_credit: 312
+
+- name: Income below 12_000
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml:9]: make unique test names

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml [hunk 17]
```diff
@@ -0,0 +1,12 @@
+description: Northwest Territories allows for this maximum amount under the cost of living tax credit.
+values:
+  2022-12-31: 942 
+metadata:
+  unit: currency-CAD
+  label: Northwest Territories cost of living tax credit max amount
+  reference:
+  - title: Government of Canada - Northwest Territories Cost of Living Credit
+    href: https://www.canada.ca/en/revenue-agency/services/forms-publications/tax-packages-years/general-income-tax-benefit-package/northwest-territories/5012-tc.html 
+  - title: Government of Canada - Income Tax Act
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:10]: could you add the section of the income tax act where the program can be found - for both yaml files

Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:10]: could you add the section of the law f.e. section 3.6 (a) or something like that

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:10]: in all yaml files

File: policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+  period: 2023
+  input:
+    province_code: NT
+    individual_net_income: 10_000
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml:5]:
```suggestion
    individual_net_income: 11_999
```

File: policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml [hunk 19]
```diff
@@ -0,0 +1,32 @@
...
+  output:
+    nt_cost_of_living_tax_credit: 260
+
+- name: Income between 12_000 and 48_000
+  period: 2023
+  input:
+    province_code: NT
+    individual_net_income: 42_000
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml:13]:
```suggestion
    individual_net_income: 12_000
```

File: policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml [hunk 20]
```diff
@@ -0,0 +1,32 @@
...
+  output:
+    nt_cost_of_living_tax_credit: 687
+
+- name: Income between 48_000 and 66_000
+  period: 2023
+  input:
+    province_code: NT
+    individual_net_income: 58_000
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml:21]:
```suggestion
    individual_net_income: 48_000
```

File: policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml [hunk 21]
```diff
@@ -0,0 +1,32 @@
...
+  output:
+    nt_cost_of_living_tax_credit: 862
+
+- name: Income above 66_000 
+  period: 2023
+  input:
+    province_code: NT
+    individual_net_income: 67_000
```
Comment (PavelMakarchuk) [inline policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml:29]:
```suggestion
    individual_net_income: 66_000
```

Comment (PavelMakarchuk) [reply policyengine_canada/tests/gov/provinces/nt/tax/income/credits/living_tax_credit/nt_cost_of_living_tax_credit.yaml:29]: we need to test edge cases

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml [hunk 22]
```diff
@@ -0,0 +1,12 @@
+description: Northwest Territories allows for this maximum amount under the cost of living tax credit.
+values:
+  2022-12-31: 942
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:3]: check 2022 amounts

Comment (Kenyaokun) [reply policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:3]: It seems that the max amount is 942 from 2005 - 2023: 
[https://www.fin.gov.nt.ca/sites/fin/files/resources/nwt_personal_income_tax_credits_english.pdf](https://www.fin.gov.nt.ca/sites/fin/files/resources/nwt_personal_income_tax_credits_english.pdf)

Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/max_amount.yaml:3]:
```suggestion
  2005-01-01: 942 
```

File: policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/rate.yaml [hunk 23]
```diff
@@ -0,0 +1,25 @@
+description: Northwest Territories extends the following cost of living credit as a fraction of the adjusted net income.
+brackets:
+  - threshold: 
+      2022-12-31: 0
```
Comment (PavelMakarchuk) [inline policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/rate.yaml:4]:
```suggestion
      2005-01-01: 0
```

Comment (PavelMakarchuk) [reply policyengine_canada/parameters/gov/provinces/nt/tax/income/credits/living_tax_credit/rate.yaml:4]: all dates

File: changelog_entry.yaml [hunk 24]
```diff
@@ -0,0 +1,4 @@
+- bump: minor
+  changes:
+    added:
+    - Northwest Territories Cost of Living Tax Credit.
```
Comment (PavelMakarchuk) [inline changelog_entry.yaml:4]:
```suggestion
    - Northwest Territories cost of living tax credit.
```
